### ISSUE TITLE
Fix TSVECTOR trigger definitions for search

### ISSUE THREAD

**Issue by jborichevskiy** [fecgov/openFEC#3798]
Searching for common names with apostrophes such as "o'hare" returns results for what seems to be all contributions containing "o" instead of the expected "o'hare".

![image](https://user-images.githubusercontent.com/7939255/58639247-91090880-82c4-11e9-90e1-ce4021440721.png)

Is there a correct way of escaping passed quotes when using the API (where the same issues seems to apply)?

**Comment by jason-upchurch** [comment]
@jborichevskiy thank you for bringing this to our attention. We will look into this issue.

**Comment by jason-upchurch** [comment]
endpoint is `/schedules/schedule_b/`

**Comment by jason-upchurch** [comment]
resultant query:
```sql
SELECT * WHERE disclosure.fec_fitem_sched_b.recipient_name_text @@ to_tsquery('o:*');
```

**Comment by jason-upchurch** [comment]
example data:

recipient_nm | recipient_name_text | old filter count | new filter count
--- | --- | --- | --- 
REPUBLICAN NATIONAL COMMITTEE - EXPENDITURES (AKA REPUBLICAN NATIONAL COMMITTEE) |  'aka':5 'c00003418':9 'committee':3,8 'expenditures':4 'national':2,7 'republican':1,6 | 1141 | 6425
 LINSLEY, JAMES A. MRS.  |  'a':3 'c00087205':5 'james':2 'linsley':1 'mrs':4 | 2 | 2
BELTRAMI COUNTY I-R ,    | 'beltrami':1 'county':2 'i':4 'i-r':3 'r':5 | 5 | 12
EMRICH, M. R.                   |  'emrich':1 'm':2 'r':3
 STACHAN, F. D. M.           |  'd':3 'f':2 'm':4 'stachan':1
 WISEMAN, EDWIN S. D.V.M.   | 'd.v.m':4 'edwin':2 's':3 'wiseman':1

**Comment by jason-upchurch** [comment]
example of 12 from BELTRAMI COUNTY I-R from above:

recipient_nm      |          recipient_name_text                    
--- | ---
 BELTRAMI COUNTY I-R ,    | 'beltrami':1 'county':2 'i':4 'i-r':3 'r':5
 BELTRAMI COUNTY IND-,  |  'beltrami':1 'county':2 'ind':3
 BELTRAMI COUNTY I-R ,    | 'beltrami':1 'county':2 'i':4 'i-r':3 'r':5
 BELTRAMI COUNTY I-R ,    | 'beltrami':1 'county':2 'i':4 'i-r':3 'r':5
 BELTRAMI COUNTY I-R , -COMMITTEE | 'beltrami':1 'committee':6 'county':2 'i':4 'i-r':3 'r':5
 BELTRAMI COUNTY I-R , -COMMITTEE | 'beltrami':1 'committee':6 'county':2 'i':4 'i-r':3 'r':5
 BELTRAMI COUNTY IR C, -OMM       |  'beltrami':1 'c':4 'county':2 'ir':3 'omm':5
 BELTRAMI COUNTY IR C, -OMM       |  'beltrami':1 'c':4 'county':2 'ir':3 'omm':5
 BELTRAMI COUNTY IR C, -OMM       |  'beltrami':1 'c':4 'county':2 'ir':3 'omm':5
 BELTRAMI COUNTY IR C, -OMM       |  'beltrami':1 'c':4 'county':2 'ir':3 'omm':5
 BELTRAMI COUNTY IR C, -OMM       |  'beltrami':1 'c':4 'county':2 'ir':3 'omm':5
 BELTRAMI COUNTY IR C, -OMMITTEE  | 'beltrami':1 'c':4 'county':2 'ir':3 'ommittee':5

**Comment by jason-upchurch** [comment]
To test:
```sql
new.recipient_name_text := to_tsvector(concat(regexp_replace(new.recipient_nm, '[^a-zA-Z0-9]', ' ')), ' ', new.clean_recipient_cmte_id);
```

**Comment by jason-upchurch** [comment]
Occurrences of `filter_fulltext_fields` to audit and determine need to update.
```
./webservices/resources/sched_f.py:38:    filter_fulltext_fields = [
./webservices/resources/sched_b.py:48:    filter_fulltext_fields = [
./webservices/resources/sched_b.py:106:    filter_fulltext_fields = [
./webservices/resources/rad_analyst.py:21:    filter_fulltext_fields = [
./webservices/resources/audit.py:29:    filter_fulltext_fields = [
./webservices/resources/audit.py:62:    filter_fulltext_fields = [
./webservices/resources/audit.py:118:    filter_fulltext_fields = [
./webservices/resources/audit.py:161:    filter_fulltext_fields = [
./webservices/resources/audit.py:168:        query = filters.filter_fulltext(models.AuditCandidateSearch.query, kwargs, self.filter_fulltext_fields)
./webservices/resources/audit.py:181:    filter_fulltext_fields = [
./webservices/resources/audit.py:188:        query = filters.filter_fulltext(models.AuditCommitteeSearch.query, kwargs, self.filter_fulltext_fields)
./webservices/resources/sched_c.py:30:    filter_fulltext_fields = [
./webservices/resources/committees.py:54:    filter_fulltext_fields = [
./webservices/resources/candidate_aggregates.py:162:    filter_fulltext_fields = [('q', models.CandidateSearch.fulltxt)]
./webservices/resources/candidate_aggregates.py:204:        query = filters.filter_fulltext(query, kwargs, self.filter_fulltext_fields)
./webservices/resources/aggregates.py:128:    filter_fulltext_fields = [
./webservices/resources/aggregates.py:154:    filter_fulltext_fields = [
./webservices/resources/aggregates.py:177:    filter_fulltext_fields = [
./webservices/resources/aggregates.py:217:    filter_fulltext_fields = [
./webservices/resources/costs.py:60:    filter_fulltext_fields = [
./webservices/resources/search.py:20:    filter_fulltext_fields = [
./webservices/resources/search.py:27:        query = filters.filter_fulltext(models.CandidateSearch.query, kwargs, self.filter_fulltext_fields)
./webservices/resources/search.py:40:    filter_fulltext_fields = [
./webservices/resources/search.py:47:        query = filters.filter_fulltext(models.CommitteeSearch.query, kwargs, self.filter_fulltext_fields)
./webservices/resources/dates.py:36:    filter_fulltext_fields = [
./webservices/resources/candidates.py:42:    filter_fulltext_fields = [('q', models.CandidateSearch.fulltxt)]
./webservices/resources/sched_d.py:42:    filter_fulltext_fields = [
./webservices/resources/sched_e.py:54:    filter_fulltext_fields = [
./webservices/resources/sched_e.py:124:    filter_fulltext_fields = [
./webservices/resources/sched_a.py:56:    filter_fulltext_fields = [
./webservices/resources/sched_a.py:155:    filter_fulltext_fields = [
./webservices/common/views.py:23:    filter_fulltext_fields = []
./webservices/common/views.py:49:        query = filters.filter_fulltext(query, kwargs, self.filter_fulltext_fields)
./tests/test_filters.py:211:            CalendarDatesView.filter_fulltext_fields
./tests/test_sched_d.py:36:    def test_filter_fulltext_field(self):
```

**Comment by jason-upchurch** [comment]
updated `disbursement_description_text` trigger definition from `./webservices/resources/sched_b.py`

**Comment by jason-upchurch** [comment]
| affected table |  column |  is_fixed | 
| ---                    | ---         | ---       |
| `fec_fitem_sched_a` | `recipient_name_text` | `T` |
| `fec_fitem_sched_a` | `disbursement_description_text` | `T` |

**Comment by jason-upchurch** [comment]
@jborichevskiy I updated the issue title based on the applicability of this issue to other searchable text. We expect as part of the overall resolution, the reported issue for "o'hare" will also be resolved.

**Comment by lbeaufort** [comment]
Occurrences of `filter_fulltext_fields` to audit and determine need to update.

| line number | `TSVECTOR`? | model |
| ---                | ---                 | --            |
| `./webservices/resources/audit.py:29:    filter_fulltext_fields = [` |  N | `AuditPrimaryCategory` | 
| `./webservices/resources/audit.py:62:    filter_fulltext_fields = [` | N |  `AuditCategory` |
| `./webservices/resources/audit.py:118:    filter_fulltext_fields = [` | Y (2) | `AuditCommitteeSearch`, `AuditCandidateSearch` |
| `./webservices/resources/audit.py:161:    filter_fulltext_fields = [` | Y (duplicate) | `AuditCandidateSearch` |
| `./webservices/resources/committees.py:54:    filter_fulltext_fields = [` | Y (1) | `CommitteeSearch`, `Committee` |
| `./webservices/resources/candidate_aggregates.py:162:    filter_fulltext_fields = [('q', models.CandidateSearch.fulltxt)]` | Y(1) | `CandidateSearch` |
| `./webservices/resources/costs.py:60:    filter_fulltext_fields = [` | Y(1) | `Electioneering` |
| `./webservices/resources/search.py:20:    filter_fulltext_fields = [` | Y(duplicate) | `CandidateSearch` |
| `./webservices/resources/search.py:40:    filter_fulltext_fields = [` | Y(duplicate) | `CommitteeSearch` |
| `./webservices/resources/dates.py:36:    filter_fulltext_fields = [` | Y(2) | `CalendarDate` |
| `./webservices/resources/candidates.py:42:    filter_fulltext_fields = [('q', models.CandidateSearch.fulltxt)]` | Y(duplicate) | `CandidateSearch` |

| affected table |  column |  is_fixed | commit |
| ---                    | ---         | ---       | --- |
| `ofec_committee_fulltext_audit_mv` | `fulltxt` |  |  |
| `ofec_candidate_fulltext_audit_mv` | `fulltxt` |  |  |
| `ofec_committee_fulltext_mv` | `fulltxt` |  |  |
| `ofec_candidate_fulltext_mv` | `fulltxt` |  |  |
| `ofec_electioneering_mv` | `purpose_description_text` |  |  |
| `ofec_dates_vw` | `summary_text` |  |  |
| `description_text` | `summary_text` |  |  |

**Comment by jason-upchurch** [comment]
Adding test coverage for each table in https://github.com/fecgov/openFEC/issues/3798#issuecomment-501311701

**Comment by leobuskin** [comment]
@jason-upchurch we can see the fix was merged but seems like it wasn't deployed to production yet, right? Because the problem still exists in production.

**Comment by PaulClark2** [comment]
@leobuskin thanks for checking in!

@Jason Upchurch’s work is scheduled to be merged to production on 7/9. We need to re-populate all the data tables to use the new logic. You can the follow that work here, https://github.com/fecgov/openFEC/issues/3836. #3836 is scheduled to be merged to production on 6/8.

**Comment by jason-upchurch** [comment]
@leobuskin @jborichevskiy cc @PaulClark2 @lbeaufort :
Update on this issue: because the search term parser was restored to its original definition that deals with chunks separated by special characters, the issue reported in this ticket should be resolved. As of Friday, performing the same query [originally reported](https://github.com/fecgov/openFEC/issues/3798#issue-450333402) returned 96 results. 

There remain some side effects associated with restoring this function which will be resolved under #3836. Thank you again for reporting this issue, and please let us know if you have any problems with the above query.
--Jason
